File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 6/25/2026, 9:18:35 AM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        for(int i=0; i<nums.size(); i++){
            if(umap.find(target - nums[i]) == umap.end()) umap[nums[i]] = i;
            else return {umap[target - nums[i]], i};
        }
        return{0, 0};
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 6/25/2026, 9:18:32 AM
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
     
        d={}
        for i,j in enumerate(nums):
            res=target-j
            if res in d:
                return [d[res],i]
            d[j]=i
        







